CMakeLists.txt

```diff
@@ -590,8 +590,20 @@ add_custom_command(
 add_custom_target(tests-jar DEPENDS ${TESTS_JAR})
 set_property(TARGET tests-jar PROPERTY JAR_FILE ${TESTS_JAR})
 
+# These flags are necessary in Java17+ in order for certain unit tests to
+# perform deep reflection on nonpublic members
+if (Java_VERSION_MAJOR VERSION_GREATER_EQUAL 17)
```

Comment:
Using `Java_VERSION_MAJOR` is not quite right because `JAVA_HOME` can be different from `TEST_JAVA_HOME`.